docs/website/contents/for-developers/CivicTime.md

```diff
@@ -0,0 +1,193 @@
+Ultimately, it depends on how quickly the net could fork to an era with a different slot duration.
+In practice, that's a matter of months on Cardano mainnet[^EmergencyRapidity].
+In theory --- assuming only that the Praos security argument holds and that honest stakeholders wouldn't vote for the hard fork unless their nodes were already ready for it --- it's one stability window less than the lower bound on the duration between the current era's voting deadline and the proposal being enacted.
+
+  - In Byron, that lower bound was originaly 2k slots = 4320 slots = 86400 seconds = one day, but was doubled to two days before the fork to Shelley happened.
+    So the lower bound on translations was at least one day ahead, since the Byron stability window was also 2k slots.
+
+  - After Byron and before Conway, it was 6k/f slots = 259200 slots = 259200 seconds = 3 days.
+    So the lower bound on translations was at least 1.5 days ahead, since the post-Byron stability window has always been 3k/f slots.
+
+  - Since Conway, it's been one epoch = 10k/f slots = 432000 slots = 432000 seconds = 5 days.
+    So the lower bound on translations is currently at least 3.5 days ahead, since the stability window is still 3k/f slots.
+
+Those lower bounds are for an arbitrary ledger state, and so they're the "worst-case".
+For specific ledger states, their detailed location in the epoch can increase the lower bound by up to one epoch, which has always been 5 days in every era so far.
+
+The limit is one stability window less the the post-voting buffer because --- assuming only the Praos security argument --- that's the upper bound on the age of a block the honest Praos node might need to discard as part of a rollback.
+In particular, until the oldest such block is after the voting deadline, a rollback could switch to a chain with a different voting outcome in that epoch.
+For example, switching from a chain in which the next era is only a few slots away to one in which it is at least another epoch away.
+
+*Aside*.
+On the other hand, when using a (non-recent) ledger state in a historical era, the HFC could theoretically use its knowledge about the upcoming eras in order to translate even further ahead.
+For example, the HFC could safely assume the one second duration for slots between some given Babbage ledger state and however many slots it might possibly take before that ledger state could transition from Babbage to Conway and from Conway to whatever comes after, because that mystery era is the first time the slot duration could change.
+The HFC does not do this in practice, since the extra complexity is not worthwhile; it's not necessary for syncing nodes to be smarter than caught-up nodes.
+
+[^EmergencyRapidity]: Technicality.
+    It could perhaps be weeks or days in an emergency, but a change to the slot duration would almost certainly be avoided in that case.
+
+## What About Chain Growth Failures?
+
+Roughly, Today's HFC was derived as follows.
```

Comment:
Maybe say that this is just a (best effort at a) guess

Reply:
Done, thanks.